src/api.py

```diff
@@ -1,14 +1,116 @@
-"""Minimum viable AgentService implementation.
-
-This will result in an agent that effectively acts like ChatGPT.
-"""
+from typing import List, Type
 
+from pydantic import Field
 from steamship.agents.functional import FunctionsBasedAgent
 from steamship.agents.llms.openai import ChatOpenAI
+from steamship.agents.mixins.transports.slack import (
+    SlackTransport,
+    SlackTransportConfig,
+)
+from steamship.agents.mixins.transports.steamship_widget import SteamshipWidgetTransport
+from steamship.agents.mixins.transports.telegram import (
+    TelegramTransport,
+    TelegramTransportConfig,
+)
+from steamship.agents.schema import Tool
 from steamship.agents.service.agent_service import AgentService
+from steamship.agents.tools.image_generation.stable_diffusion import StableDiffusionTool
+from steamship.invocable import Config
+from steamship.utils.repl import AgentREPL
+
+
+class BasicAgent(AgentService):
```

Comment:
BasicAgent is a name both here and in example_agents and they currently have quite a bit of overlap.  Could this get pared down or could we remove basic_agent.py from the examples folder?

Reply:
Yeah this is a good point. 

It has to be here in `api.py` since that's what our deployment service looks for, but we could just import one of the examples and re-export it here -- I'll do that since that results in only one copy of the code.